core/src/client/async_client/mod.rs

```diff
@@ -740,81 +668,188 @@ async fn background_task<S, R>(
 	// This implies that either messages are received (both front or backend),
 	// or the submitted ping timer expires (if provided).
 	let mut message_fut = future::select(frontend.next(), backend_event.next());
-	let mut exit_or_ping_fut = future::select(on_exit, ping_fut(ping_interval));
+	let closed = close_tx.closed();
+
+	tokio::pin!(closed);
 
-	loop {
-		match future::select(message_fut, exit_or_ping_fut).await {
+	let res = loop {
+		match future::select(message_fut, closed).await {
 			// Message received from the frontend.
-			Either::Left((Either::Left((frontend_value, backend)), exit_or_ping)) => {
+			Either::Left((Either::Left((frontend_value, backend)), c)) => {
 				let frontend_value = if let Some(value) = frontend_value {
 					value
 				} else {
 					// User dropped the sender side of the channel.
 					// There is nothing to do just terminate.
 					tracing::debug!("[backend]: Client dropped");
-					break;
+					break Ok(());
 				};
 
-				handle_frontend_messages(
-					frontend_value,
-					&mut manager,
-					&mut sender,
-					max_buffer_capacity_per_subscription,
-				)
-				.await;
+				match frontend_value {
+					FrontToBack::Batch(batch) => {
+						if let Err(send_back) = manager.insert_pending_batch(batch.ids.clone(), batch.send_back) {
+							tracing::warn!("[backend]: Batch request already pending: {:?}", batch.ids);
+							let _ =
```

Comment:
Would a tracing log be of use here? Or could we deduct elsewhere that the batch channel was dropped?

Reply:
The channel is only closed if the batch request was dropped but sure we could add some debug logs for that

Reply:
ah, I'll ignore it in this PR because I would like to do it for all operations or none :P